### DISCUSSION TITLE
Make Explicit or Better Document `seq!` Reffering Behavior

### DISCUSSION THREAD

**Discussion by bbb651** [winnow-rs/winnow#534] [Ideas]
(moved from [#406](https://github.com/winnow-rs/winnow/issues/406#issuecomment-2143308304))

I'm just starting out with winnow and I've spent an embarrassingly long amount of time getting unreadable type error from the compiler because I didn't know about the behavior of referring to other struct fields and it doesn't seem to be documented anywhere.
My code was something like:

```rs
seq! { Foo {
    identifier: identifier,
    // ...
}}
```

and I was trying to use identifier thinking it was the parser and not the field. Not sure if it's possible with `macro_rules!` or worth it but having it prefixed with `let` or replacing `:` with `=` to explicitly have these references could really help, but at the very least document this properly.

**Comment by epage** [comment]
> `seq!` is trying to mirror the struct initialization syntax within Rust today which is why `:` is used.  I hadn't thought of shadowing issues like this that arise from the discrepancy of behavior.

At the moment, my feeling is there is no good answer and I'd prefer to keep what we have.

I have posted #535 to see what tweaks can be made to clarify what is happening
